test/e2e_node/util.go

```diff
@@ -171,7 +171,7 @@ func getCurrentKubeletConfig(ctx context.Context) (*kubeletconfig.KubeletConfigu
 	return e2enodekubelet.GetCurrentKubeletConfig(ctx, framework.TestContext.NodeName, "", false, framework.TestContext.StandaloneMode)
 }
 
-func cleanupPods(f *framework.Framework) {
+func addAfterEachForCleaningUpPods(f *framework.Framework) {
```

Comment:
sorry for being late to the party (timezone fun) and thanks @dims for the quick and nice fix. Out of experience this will help but it seems it will likely hit two known pain points:
1. deletion should preferably be done concurrently (as in `go DeleteSync` + `sync.WaitGroup`)
2. we likely need `waitForContainerRemoval` after each pod deletion

Reply:
@ffromani we are doing `DeleteSync` in https://github.com/kubernetes/kubernetes/blob/0c8b3e5f305bf2bf56d47019199b81330d90c2c3/test/e2e_node/util.go#L174-L187

we should enhance the AfterRach in `addAfterEachForCleaningUpPods` to do what you suggest if what we have there is not enough.